vcr/issuer/issuer.go

```diff
@@ -47,36 +47,42 @@ import (
 var TimeFunc = time.Now
 
 // NewIssuer creates a new issuer which implements the Issuer interface.
-// If oidcIssuer is nil, it won't try to issue over OIDC4VCI.
+// If openidIssuerFn is nil, it won't try to issue over OIDC4VCI.
 // It needs types.Writer since issued credentials need to be in the general VCR store,
 // since that normally happens through receiving the just-issued credential over the network,
 // but that doesn't happen when issuing over OIDC4VCI. Thus, it needs to explicitly save it to the VCR store when issuing over OIDC4VCI.
 // See https://github.com/nuts-foundation/nuts-node/issues/2063
-func NewIssuer(store Store, vcrStore types.Writer, networkPublisher Publisher, oidcIssuer openid4vci.Issuer, docResolver vdr.DocResolver, keyStore crypto.KeyStore, jsonldManager jsonld.JSONLD, trustConfig *trust.Config) Issuer {
+func NewIssuer(store Store, vcrStore types.Writer, networkPublisher Publisher,
+	openidIssuerFn func(ctx context.Context, id did.DID) (openid4vci.Issuer, error),
+	docResolver vdr.DocResolver, keyStore crypto.KeyStore, jsonldManager jsonld.JSONLD, trustConfig *trust.Config,
+) Issuer {
 	resolver := vdrKeyResolver{docResolver: docResolver, keyResolver: keyStore}
 	return &issuer{
 		store:            store,
 		networkPublisher: networkPublisher,
-		oidcIssuer:       oidcIssuer,
-		keyResolver:      resolver,
-		keyStore:         keyStore,
-		serviceResolver:  didservice.NewServiceResolver(docResolver),
-		jsonldManager:    jsonldManager,
-		trustConfig:      trustConfig,
-		vcrStore:         vcrStore,
+		openidIssuerFn:   openidIssuerFn,
+		walletResolver: service.DIDIdentifierResolver{
+			ServiceResolver: didservice.NewServiceResolver(docResolver),
+		},
+		keyResolver:   resolver,
+		keyStore:      keyStore,
+		jsonldManager: jsonldManager,
+		trustConfig:   trustConfig,
+		vcrStore:      vcrStore,
 	}
 }
 
 type issuer struct {
 	store            Store
 	networkPublisher Publisher
-	oidcIssuer       openid4vci.Issuer
+	openidIssuerFn   func(ctx context.Context, id did.DID) (openid4vci.Issuer, error)
```

Comment:
Code smell, this might be hiding some spaghetti-like construction.
The referenced method uses the `walletResolver` and the `DocumentOwner`. You already have the walletResolver, only need the DocumentOwner service. Then there's no need to back-reference to the VCR.